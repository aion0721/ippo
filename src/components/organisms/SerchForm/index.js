import React from 'react';
import propTypes from 'prop-types';
import { Button, InputGroup } from '@blueprintjs/core';

class SearchForm extends React.Component {
  state = { searchWord: '' };

  onChangeSearchWord = event => this.setState({ searchWord: event.target.value });

  onSubmit = () => {
    this.props.search(this.state.searchWord);
  };

  render() {
    const { searchWord } = this.state;
    return (
      <InputGroup
        onChange={this.onChangeSearchWord}
        value={searchWord}
        leftIcon="search"
        rightElement={<Button onClick={this.onSubmit} icon="arrow-right" minimal />}
        large
      />
    );
  }
}

SearchForm.displayName = 'SearchForm';

SearchForm.propTypes = {
  search: propTypes.func.isRequired,
};

export default SearchForm;
