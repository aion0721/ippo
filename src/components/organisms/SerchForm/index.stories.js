import React from 'react';
import { storiesOf } from '@storybook/react';
import SearchForm from '.';

const stories = storiesOf('organisms/SearchForm', module);

stories.add('通常パターン', () => <SearchForm search={val => alert(val)} />);
