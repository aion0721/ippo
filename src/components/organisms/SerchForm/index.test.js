import React from 'react';
import { snapshot } from 'test/helpers';
import SerchForm from '.';

const props = {
  search: jest.fn(),
};

snapshot('SerchForm/nomal', <SerchForm {...props} />);
