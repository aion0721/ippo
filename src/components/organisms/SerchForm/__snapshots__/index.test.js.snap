// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: SerchForm/nomal 1`] = `
<SearchForm
  search={[MockFunction]}
>
  <Blueprint3.InputGroup
    large={true}
    leftIcon="search"
    onChange={[Function]}
    rightElement={
      <Blueprint3.Button
        icon="arrow-right"
        minimal={true}
        onClick={[Function]}
      />
    }
    value=""
  >
    <div
      className="bp3-input-group bp3-large"
    >
      <Blueprint3.Icon
        icon="search"
      >
        <span
          className="bp3-icon bp3-icon-search"
          icon="search"
        >
          <svg
            data-icon="search"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              search
            </desc>
            <path
              d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              fillRule="evenodd"
              key="0"
            />
          </svg>
        </span>
      </Blueprint3.Icon>
      <input
        className="bp3-input"
        onChange={[Function]}
        style={
          Object {
            "paddingRight": 0,
          }
        }
        type="text"
        value=""
      />
      <span
        className="bp3-input-action"
      >
        <Blueprint3.Button
          icon="arrow-right"
          minimal={true}
          onClick={[Function]}
        >
          <button
            className="bp3-button bp3-minimal"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <Blueprint3.Icon
              icon="arrow-right"
              key="leftIcon"
            >
              <span
                className="bp3-icon bp3-icon-arrow-right"
                icon="arrow-right"
              >
                <svg
                  data-icon="arrow-right"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    arrow-right
                  </desc>
                  <path
                    d="M14.7 7.29l-5-5a.965.965 0 0 0-.71-.3 1.003 1.003 0 0 0-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 0 0 1.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
            <Blueprint3.Icon
              key="rightIcon"
            />
          </button>
        </Blueprint3.Button>
      </span>
    </div>
  </Blueprint3.InputGroup>
</SearchForm>
`;
